src/trans/gpu/internal/ftdir_mod.F90

```diff
@@ -10,19 +10,35 @@
 !
 
 MODULE FTDIR_MOD
+  USE BUFFERED_ALLOCATOR_MOD ,ONLY : ALLOCATION_RESERVATION_HANDLE
   IMPLICIT NONE
 
   PRIVATE
   PUBLIC :: FTDIR, FTDIR_HANDLE, PREPARE_FTDIR
 
   TYPE FTDIR_HANDLE
+    TYPE(ALLOCATION_RESERVATION_HANDLE) :: HREEL_COMPLEX
   END TYPE
 CONTAINS
 
-  FUNCTION PREPARE_FTDIR() RESULT(HFTDIR)
+  FUNCTION PREPARE_FTDIR(ALLOCATOR,KF_FS) RESULT(HFTDIR)
+    USE PARKIND_ECTRANS,        ONLY: JPIM, JPRBT
+    USE TPM_DISTR,              ONLY: D
+    USE BUFFERED_ALLOCATOR_MOD, ONLY: BUFFERED_ALLOCATOR, RESERVE
+    USE ISO_C_BINDING,          ONLY: C_SIZE_T
+
     IMPLICIT NONE
+
+    TYPE(BUFFERED_ALLOCATOR), INTENT(INOUT) :: ALLOCATOR
+    INTEGER(KIND=JPIM), INTENT(IN) :: KF_FS
     TYPE(FTDIR_HANDLE) :: HFTDIR
-  END FUNCTION
+
+    REAL(KIND=JPRBT) :: DUMMY
+
+#ifndef IN_PLACE_FFT
+    HFTDIR%HREEL_COMPLEX = RESERVE(ALLOCATOR, INT(KF_FS*D%NLENGTF*SIZEOF(DUMMY), KIND=C_SIZE_T))
```

Comment:
`SIZEOF` is a non-standard extension. Some compilers could complain.
Standard in F2008 is `STORAGE_SIZE`, which gives you `bits` (not bytes!) and `C_SIZEOF`, which gives you `bytes`.

Reply:
`SIZEOF` occurs in a few places in the GPU tree. @lukasm91 any strong feelings about switching to `8*STORAGE_SIZE(DUMMY)`?